lib/service/service.go

```diff
@@ -1240,9 +1254,8 @@ func (process *TeleportProcess) initAuthService() error {
 			process.log.Warn(warningMessage)
 		}
 
-		auditConfig := cfg.Auth.AuditConfig
 		uploadHandler, err = initUploadHandler(
-			process.ExitContext(), auditConfig, filepath.Join(cfg.DataDir, teleport.LogsDir))
+			process.ExitContext(), cfg.Auth.AuditConfig, filepath.Join(cfg.DataDir, teleport.LogsDir), cfg.FIPS)
```

Comment:
Can we tuck the "FIPS" flag (i.e. `cfg.FIPS`) inside `cfg.Auth.AuditConfig` so we don't have to pass 5th parameter everywhere?

It would make sense I think - wherever AuditConfig is initialized, we would init it with FIPS flag which would indicate that auditing should be FIPS compatible.

Reply:
This bit led to an exploration of where `ClusterAuditConfig` comes from and it's a combination of custom interfaces built on top of `protobuf` and `ClusterAuditConfig`. Protobuf can't easily represent a pointer to a boolean, so I'll need to declare specific states for it. That's ok, they can just mirror the AWS states. It just felt a little scope creepy when I was first looking at it, but you make a good point here that it indicates that auditing should be FIPS compatible.